src/main/java/com/commercetools/sync/shoppinglists/helpers/ShoppingListBatchValidator.java

```diff
@@ -0,0 +1,222 @@
+                                                              final int variantPosition,
+                                                              @Nonnull final String shoppingListDraftKey) {
+        final List<String> errorMessages = new ArrayList<>();
+        if (lineItemDraft != null) {
+            if (isBlank(lineItemDraft.getSku())) {
+                errorMessages.add(format(LINE_ITEM_DRAFT_SKU_NOT_SET, variantPosition, shoppingListDraftKey));
+            }
+        } else {
+            errorMessages.add(format(LINE_ITEM_DRAFT_IS_NULL, variantPosition, shoppingListDraftKey));
+        }
+        return errorMessages;
+    }
+
+    @Nonnull
+    private List<String> getTextLineItemDraftErrorsInAllTextLineItems(
+        @Nullable final TextLineItemDraft textLineItemDraft, final int variantPosition,
+        @Nonnull final String shoppingListDraftKey) {
+        final List<String> errorMessages = new ArrayList<>();
+        if (textLineItemDraft != null) {
+            if (textLineItemDraft.getName() == null || textLineItemDraft.getName().getLocales().isEmpty()) {
+                errorMessages.add(format(TEXT_LINE_ITEM_DRAFT_NAME_NOT_SET, variantPosition, shoppingListDraftKey));
+            }
+        } else {
+            errorMessages.add(format(TEXT_LINE_ITEM_DRAFT_IS_NULL, variantPosition, shoppingListDraftKey));
+        }
+        return errorMessages;
+    }
+
+    private void collectReferencedKeys(
+        @Nonnull final ReferencedKeys referencedKeys,
+        @Nonnull final ShoppingListDraft shoppingListDraft) {
+
+        collectReferencedKeyFromResourceIdentifier(shoppingListDraft.getCustomer(),
+            referencedKeys.customerKeys::add);
+        collectReferencedKeyFromCustomFieldsDraft(shoppingListDraft.getCustom(),
+            referencedKeys.typeKeys::add);
+        collectReferencedKeysInLineItems(referencedKeys, shoppingListDraft);
+        collectReferencedKeysInTextLineItems(referencedKeys, shoppingListDraft);
+
+    }
+
+    private void collectReferencedKeysInLineItems(
+        @Nonnull final ReferencedKeys referencedKeys,
+        @Nonnull final ShoppingListDraft shoppingListDraft) {
+
+        if (shoppingListDraft.getLineItems() == null) {
+            return;
+        }
+
+        shoppingListDraft
+            .getLineItems()
+            .stream()
+            .filter(Objects::nonNull)
+            .forEach(lineItemDraft -> {
+                collectReferencedKeyFromCustomFieldsDraft(lineItemDraft.getCustom(),
+                    referencedKeys.typeKeys::add);
+            });
+    }
+
+    private void collectReferencedKeysInTextLineItems(
+        @Nonnull final ReferencedKeys referencedKeys,
+        @Nonnull final ShoppingListDraft shoppingListDraft) {
+
+        if (shoppingListDraft.getTextLineItems() == null) {
+            return;
+        }
+
+        shoppingListDraft
+            .getTextLineItems()
+            .stream()
+            .filter(Objects::nonNull)
```

Comment:
same here, not sure if you need this check 🤔

Reply:
0c45ae6